feat(docs): Improve docs for dynamic sampling biases. [TET-559] (#5824)
getsentry/sentry-docs: src/docs/product/data-management-settings/dynamic-sampling/index.mdx

| Prioritize | Ignore |
|------------|--------|
| Latest Release | Health Checks |
| Dev Environments | |

<!-- prettier-ignore-end -->

### Latest Release

> Every time you create a new release, there is a good chance you will want to get more visibility during the early adoption phase. As a result, to have the ability to catch new issues as soon as your release is being adopted, we prioritize data from the latest release.

Reviewer: ```suggestion
When you create a new release, you'll likely want to have more visibility during the early adoption phase. To improve your ability to catch new issues as soon as your release is being adopted, we prioritize data from the latest release.
```

Author: Also amazing!

> _While prioritizing data from the latest release, we will also take into consideration your environment because sometimes you might keep the same release and only change the environment. In that case, we also assume you would expect to get more visibility on those transactions._

Reviewer: We use italics to emphasize specific words that are new to the user. Did you want this to be a note? It seems like having it as plain text works fine.

```suggestion
While prioritizing data from the latest release, we also take into consideration your environment because you might deploy the same release in different environments. If we see this, we'll assume you want to have more visibility for those transactions as well.
```

Author: Thanks a lot! I love how smooth it sounds, way better!
